internal/states/remote/state.go

```diff
@@ -175,13 +179,15 @@ func (s *State) PersistState(schemas *terraform.Schemas) error {
 		if err != nil {
 			return fmt.Errorf("failed checking for existing remote state: %s", err)
 		}
+		log.Printf("[DEBUG] states/remote: after refresh, state read serial is: %d; serial is: %d", s.readSerial, s.serial)
+		log.Printf("[DEBUG] states/remote: after refresh, state read lineage is: %s; lineage is: %s", s.readLineage, s.lineage)
 		if s.lineage == "" { // indicates that no state snapshot is present yet
 			lineage, err := uuid.GenerateUUID()
 			if err != nil {
 				return fmt.Errorf("failed to generate initial lineage: %v", err)
 			}
 			s.lineage = lineage
-			s.serial = 0
+			s.serial = 1
```

Comment:
I might suggest changing this to match what is done in other places in the codebase:

```suggestion
			s.serial++
```

Reply:
@brandonc, can you confirm if this would affect cloud at all, or does the server itself prevent this from being applied?

Reply:
The Terraform Cloud platform will allow this to be applied, as long as it is incrementing. I duplicated the fix as https://github.com/hashicorp/terraform/pull/32647 and smoke tested it in Terraform Cloud. Thanks for the mention